library/ceph_volume.py

```diff
@@ -0,0 +1,240 @@
+    data: /dev/sdc
+
+- name: set up a bluestore osd with an lv for data and partitions for block.wal and block.db
+  ceph_volume:
+    objectstore: bluestore
+    data: data-lv
+    data_vg: data-vg
+    db: /dev/sdc1
+    wal: /dev/sdc2
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def get_data(data, data_vg):
+    if data_vg:
+        data = "{0}/{1}".format(data_vg, data)
+    return data
+
+
+def get_journal(journal, journal_vg):
+    if journal_vg:
+        journal = "{0}/{1}".format(journal_vg, journal)
+    return journal
+
+
+def get_db(db, db_vg):
+    if db_vg:
+        db = "{0}/{1}".format(db_vg, db)
+    return db
+
+
+def get_wal(wal, wal_vg):
+    if wal_vg:
+        wal = "{0}/{1}".format(wal_vg, wal)
+    return wal
+
+
+def run_module():
+    module_args = dict(
+        cluster=dict(type='str', required=False, default='ceph'),
+        subcommand=dict(type='str', required=False, default='lvm'),
+        objectstore=dict(type='str', required=True),
+        data=dict(type='str', required=True),
+        data_vg=dict(type='str', required=False),
+        journal=dict(type='str', required=False),
+        journal_vg=dict(type='str', required=False),
+        db=dict(type='str', required=False),
+        db_vg=dict(type='str', required=False),
+        wal=dict(type='str', required=False),
+        wal_vg=dict(type='str', required=False),
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    cluster = module.params['cluster']
+    subcommand = module.params['subcommand']
+    objectstore = module.params['objectstore']
+    data = module.params['data']
+    data_vg = module.params.get('data_vg', None)
+    journal = module.params.get('journal', None)
+    journal_vg = module.params.get('journal_vg', None)
+    db = module.params.get('db', None)
+    db_vg = module.params.get('db_vg', None)
+    wal = module.params.get('wal', None)
+    wal_vg = module.params.get('wal_vg', None)
+
+    cmd = [
+        'ceph-volume',
+        '--cluster',
+        cluster,
+        subcommand,
+        'create',
+        '--%s' % objectstore,
+        '--data',
+    ]
+
+    data = get_data(data, data_vg)
+    cmd.append(data)
+
+    if journal:
+        journal = get_journal(journal, journal_vg)
+        cmd.extend(["--journal", journal])
+
+    if db:
+        db = get_db(db, db_vg)
+        cmd.extend(["--block.db", db])
+
+    if wal:
+        wal = get_wal(wal, wal_vg)
+        cmd.extend(["--block.wal", wal])
+
+    result = dict(
+        changed=False,
+        cmd=cmd,
+        stdout='',
+        stderr='',
+        rc='',
+        start='',
+        end='',
+        delta='',
+    )
+
+    if module.check_mode:
+        return result
+
+    # check to see if osd already exists
+    # FIXME: this does not work when data is a raw device
+    rc, out, err = module.run_command(["ceph-volume", "lvm", "list", data], encoding=None)
```

Comment:
I don't see anything here that enables the DEBUG mode in ceph-volume, which I think is a must. If we aren't planning on making it the default, at least we should provide a switch to enable it

Reply:
Ansible already has support for this so I just left it on the task itself https://github.com/ceph/ceph-ansible/pull/2227/files#diff-0f31463b61560d27879319f18129b679R15

You'd rather have a ``debug`` param on the ``ceph_volume`` module itself?

Reply:
No, sorry about that, I didn't catch the env passing when calling the module